### DISCUSSION TITLE
Initialiser un champ dans le formulaire

### DISCUSSION THREAD

**Discussion by ChrisThe01** [MTESSDev/FRW#214] [Annonces]
Est-ce qu'il est possible d'initialiser un champ directement dans un formulaire ?

**Comment by loima02** [comment]
Bonjour,

Vous trouverez l'information dans la section "Pré-remplissage" de notre documentation
Réf : https://github.com/MTESSDev/FRW/blob/main/Documentation/pre-remplissage.md

Vous pouvez tester vos données grâce à la section "Données en entrée (pré-remplissage)" de nos outils de développement.
Réf : https://github.com/MTESSDev/FRW/blob/main/Documentation/outils-developpement.md

**Comment by ChrisThe01** [comment]
Je voudrais que l'information soit saisie directement dans le tableau dans le formulaire, sans préremplissage.

**Comment by j0ber** [comment]
Soyez plus précis avec votre exemple... Ça ne fait pas de sens... Initialisé avec quelle valeur? Si ce n'est pas prérempli la valeur à utiliser pour l'initialisation est laquelle?  Est-ce pour le total des champs d'un tableau de la discussion #213 ?

**Comment by ChrisThe01** [comment]
Je voudrais un champ (ou variable), par exemple var_xxx, dont je voudrais **assigner** une valeur. Ce champ ne sera pas visible à l'utilisateur. Je ne voudrais pas être obliger d'utiliser un préremplisage pour assigner une valeur à un champ

var_xxx = "FlagA"

Est-ce que ça l'a plus de sens ? Sinon, contacte moi au pire.

PS : Cette discussion n'a pas rapport avec la discussion 213.

Merci,

**Comment by j0ber** [comment]
Est-ce possible de connaître votre cas d'utilisation concret? On parle de simplement avoir une valeur par défaut pour un champ? ex. un champ texte avec la valeur "Salut" dans son contenu par défaut?

**Comment by ChrisThe01** [comment]
Oui

**Comment by j0ber** [comment]
Il suffit d'utiliser la propriété "value". Elle permet de définir la valeur initiale (par défaut) d'un champ : 

````YAML
- type: text
  name: monChamp1
  label:
    fr: Un libellé très pertinent
  value: "Bonjour!"
````

Un exemple a été ajouté dans le guide de configuration https://formulaires.it.mtess.gouv.qc.ca/fr/Form/7/P700U/0/N/#p=controlesBase  voir exemple 5.
